tools/autograd/derivatives.yaml

```diff
@@ -383,6 +383,11 @@
   self: grad
   result: auto_linear
 
+- name: _to_copy(Tensor self, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None, bool non_blocking=False, MemoryFormat? memory_format=None) -> Tensor
+  self: _to_copy_backward(grad, self.options())
```

Comment:
Can't this one be inline? Isn't there a to() overload that takes options as input?

Reply:
`_to_copy_backward` calls `_to_copy` instead of `to()` for efficiency. There is no overload of `_to_copy` that takes options as input. `_to_copy` with arguments is a bit long so that's why this isn't inline

Reply:
> _to_copy instead of to() for efficiency

Is the difference really that big?
Also maybe the gradient already has the right properties and you don't want to copy it?

Reply:
> Is the difference really that big?

The difference is another redispatch, so it's really not that big.

> Also maybe the gradient already has the right properties and you don't want to copy it?

This can't happen if the user calls `tensor.to(...)`, but it could happen if a user uses `_to_copy` directly. In the future these `_copy` APIs may be exposed to users, so let's replace this with `::to`

Reply:
> This can't happen if the user calls tensor.to(...)

Really? I could see a case where we clone in the forward to satisfy the requires layout, but the layout is already correct in the backward. no?

Reply:
Good point, the layout of gradients is not always the same as the tensor (though we try to make it so and there are some special cases about leaf nodes and their .grad fields)